### PR TITLE
"Rename" `NetworkPolicy`: `allow-to-seed-apiserver` -> `allow-to-runtime-apiserver`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area networking security
/kind enhancement

**What this PR does / why we need it**:
This PR makes `gardenlet'`s `NetworkPolicy` controller not only deploy the `allow-to-seed-apiserver` `NetworkPolicy` but the same one with the new `allow-to-runtime-apiserver` name. This is needed since the controller is planned to be also used by `gardener-operator` in garden!=seed clusters.

In addition, the `istio-system` namespace in seed clusters is now labeled with `gardener.cloud/role=istio-system`, and all `istio-ingress*` namespaces are now labeled with `gardener.cloud/role=istio-ingress` for easier identification.

**Which issue(s) this PR fixes**:
Part of #7352 

**Special notes for your reviewer**:
In the future, the `gardenlet`'s `NetworkPolicy` controller will manage more `NetworkPolicy` resources.

**Release note**:

```breaking operator
The configuration for `gardenlet`'s `NetworkPolicy` controller must now be submitted via `.controllers.networkPolicy` instead of `controllers.seedAPIServerNetworkPolicy`.
```
```breaking dependency
The `allow-to-seed-apiserver` `NetworkPolicy` is now deprecated and replaced by the new `allow-to-runtime-apiserver` `NetworkPolicy`. Components running in the seed cluster that need to talk to the `kube-apiserver` should be labeled with `networking.gardener.cloud/to-runtime-apiserver=allowed`.
```
```feature operator
The `istio-system` namespace in seed clusters is now labeled with `gardener.cloud/role=istio-system`. All `istio-ingress*` namespaces are now labeled with `gardener.cloud/role=istio-ingress`.
```

### CODE REVIEW

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
I added a new commit [eb09360](https://github.com/gardener/gardener/pull/7389/commits/eb09360016c37c73fdf86e9aec551fc58484c358) to simplify the logic in one of the reconciler's mapping functions.

File: pkg/operation/botanist/component/istio/istiod_test.go [hunk 1]
```diff
@@ -2263,6 +2263,7 @@ spec:
 				HaveKeyWithValue("istio-operator-managed", "Reconcile"),
 				HaveKeyWithValue("istio-injection", "disabled"),
 				HaveKeyWithValue("high-availability-config.resources.gardener.cloud/consider", "true"),
+				HaveKeyWithValue("gardener.cloud/role", "istio-ingress"),
```
Comment (timuthy) [inline pkg/operation/botanist/component/istio/istiod_test.go:2267]: Can you add the same to the `"deploys istiod namespace"` test?

File: pkg/gardenlet/controller/networkpolicy/add.go [hunk 2]
```diff
@@ -126,19 +127,22 @@ func (r *Reconciler) NetworkPolicyPredicate() predicate.Predicate {
...
+	if err := r.RuntimeClient.List(ctx, namespaceList); err != nil {
+		log.Error(err, "Unable to list all namespaces")
+		return nil
+	}
+
+	namespaceNames := sets.NewString()
+	for _, namespace := range namespaceList.Items {
+		if labelsMatchAnySelector(namespace.Labels, selectors) {
+			namespaceNames.Insert(namespace.Name)
+			continue
```
Comment (timuthy) [inline pkg/gardenlet/controller/networkpolicy/add.go]:
```suggestion
```

**Review by timuthy — COMMENTED** [review]
Nice, two nits from my side. I destructively tested the changes in my local setup w/o any issues ✅

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**